packages/tools/src/tools/ZoomTool.ts

```diff
@@ -46,6 +48,74 @@ class ZoomTool extends BaseTool {
     }
     this.mouseDragCallback = this._dragCallback.bind(this);
   }
+  // Helper to transform a normal by a 3x3 matrix
+  _transformNormal(normal: Types.Point3, mat: number[]): Types.Point3 {
+    return [
+      mat[0] * normal[0] + mat[3] * normal[1] + mat[6] * normal[2],
+      mat[1] * normal[0] + mat[4] * normal[1] + mat[7] * normal[2],
+      mat[2] * normal[0] + mat[5] * normal[1] + mat[8] * normal[2],
+    ];
+  }
+
+  _updateClippingPlanes(viewport) {
```

Comment:
Also, you have the same code across different tools; we should always extract it into a utility and reuse it. But I think these things should not live in Zoom, rotate, etc, only in the volume cropping tool. If existing events are not enough, we could think about adding events for these tools and indirectly using them from the volume cropping domain.